### ISSUE TITLE
`format_yaml()` method support for JSON numbers

### ISSUE THREAD

**Issue by mihaitodor** [Jeffail/benthos#1010]
Currently, `yaml.v3` marshals `json.Number` as string: https://github.com/go-yaml/yaml/issues/807 while `yaml.v2` handles them properly https://github.com/go-yaml/yaml/pull/414.

Will wait a bit for feedback on https://github.com/go-yaml/yaml/issues/807 before pushing any workarounds.

**Comment by twmb** [comment]
The upstream issue at https://github.com/go-yaml/yaml/issues/807 remains open with no activity. Closing as this is blocked on upstream with no indication of resolution. If the upstream fix lands or a workaround is implemented, feel free to open a new issue or PR.
